multiversx_sdk/transfers/transfer_transactions_factory.py

```diff
@@ -58,6 +64,10 @@ def create_transaction_for_esdt_token_transfer(
         else:
             data_parts, extra_gas_for_transfer, receiver = self._multi_transfer(sender, receiver, token_transfers)
 
+        if data:
+            serializer = Serializer()
+            data_parts.append(serializer.serialize([StringValue(data.decode())]))
```

Comment:
Since we're in this place (custom message for token transfers), can also simply be `data.hex()`.

Can also stay as it is, though.